File: src/main/java/org/micoli/php/openAPI/OpenAPIPathElementDTO.java
package org.micoli.php.openAPI;

import com.intellij.psi.PsiElement;
import java.util.List;
import org.jetbrains.annotations.NotNull;
import org.micoli.php.symfony.list.SearchableRecord;

public record OpenAPIPathElementDTO(
        @NotNull String rootPath,
        @NotNull String uri,
        @NotNull String method,
        @NotNull String description,
        @NotNull String operationId,
        PsiElement element)
        implements SearchableRecord {

    @Override
    public List<@NotNull String> getSearchString() {
        return List.of(uri, method, description, operationId, rootPath);
    }
}

File: src/main/kotlin/org/micoli/php/openAPI/OpenAPIPathElementDTO.kt
package org.micoli.php.openAPI

import com.intellij.psi.PsiElement
import kotlinx.collections.immutable.ImmutableList
import kotlinx.collections.immutable.persistentListOf
import org.micoli.php.symfony.list.SearchableRecord

@JvmRecord
data class OpenAPIPathElementDTO(
    val rootPath: String,
    val uri: String,
    val method: String,
    val description: String,
    val operationId: String,
    val element: PsiElement?
) : SearchableRecord {
    override fun getSearchString(): ImmutableList<String> {
        return persistentListOf(uri, method, description, operationId, rootPath)
    }
}
